### DISCUSSION TITLE
How to make a existing package/component compatible with web using react-native-web?

### DISCUSSION THREAD

**Discussion by ghost** [necolas/react-native-web#2101] [Q&A]
Hi, 
I'm very new to react-native-web, I need to know how react-native-web works. 
Suppose I have a react-native components which doesn't have web support. How can I add web support to that component using react-native-web.

Thanks in advance

**Comment by NarayanaRao35** [comment]
here is some useful links 
https://blog.logrocket.com/react-native-web-complete-tutorial/ 
https://necolas.github.io/react-native-web/docs/
https://dev.to/brunolemos/tutorial-100-code-sharing-between-ios-android--web-using-react-native-web-andmonorepo-4pej

**Comment by nandorojo** [comment]
You can use `filename.web.tsx` files to make `filename.tsx` import a different file on web altogether (including that file's dependencies). You can also use the `Platform.select` or `Platform.OS` APIs from react native.

I have this in a video here: https://youtu.be/0lnbdRweJtA?t=1315
